### PR TITLE
Improvements for PICARD-2197 alias sort name lookup

### PR DESCRIPTION
**Pull request by zas**
Suggested improvements for #3157:

1. **Add test for locale-preferred sort name selection** — exercises the `translate_artist_names=True` path in `_select_sort_name_from_aliases` with multiple locale aliases.

2. **Add docstring and edge case comment** — documents the function's purpose, fallback behavior, and the empty sort-name edge case.

3. **Skip alias lookup when credited name equals canonical name** — avoids unnecessary work in both `artist_credit_from_node` and `_relations_to_metadata_target_type_artist` when the credit is the same as the artist's canonical name.

### CODE REVIEW

File: picard/mbjson.py [hunk 1]
```diff
@@ -644,7 +651,10 @@ def artist_credit_from_node(node: list[Node]) -> ArtistCreditInfo:
         has_translation = translated_alias.name != artist['name']
         if not has_translation and use_credited_as and 'name' in artist_info:
             name = artist_info['name']
-            sort_name = _select_sort_name_from_aliases(artist, name, config=config)
+            if name != artist['name']:
```
Comment (phw) [inline picard/mbjson.py:654]: That's not necessary, this is already checked above (`has_translation`)

**Review by phw — CHANGES_REQUESTED** [review]

**Comment by phw** [comment]
Cherry picked with change directly into https://github.com/metabrainz/picard/pull/3157
